### PR TITLE
Updated Kotlin to 1.5.10, Android SDK/Gradle Plugin, dependencies, Gradle wrapper

### PR DESCRIPTION
**Pull request by shepeliev**
This is a fork of PR #178, but Android tests fixed and the latest versions for Kotlin, AGP, coroutines.

### CODE REVIEW

File: firebase-common/build.gradle.kts [hunk 1]
```diff
@@ -9,7 +9,7 @@ version = project.property("firebase-common.version") as String
 plugins {
     id("com.android.library")
     kotlin("multiplatform")
-    kotlin("plugin.serialization") version "1.4.31"
+    kotlin("plugin.serialization") version "1.5.0"
```
Comment (suntrix) [inline firebase-common/build.gradle.kts:12]: 1.5.10

File: firebase-firestore/build.gradle.kts [hunk 2]
```diff
@@ -9,7 +9,7 @@ version = project.property("firebase-firestore.version") as String
 plugins {
     id("com.android.library")
     kotlin("multiplatform")
-    kotlin("plugin.serialization") version "1.4.31"
+    kotlin("plugin.serialization") version "1.5.0"
```
Comment (suntrix) [inline firebase-firestore/build.gradle.kts:12]: 1.5.10

File: firebase-common/build.gradle.kts (hunk 1 above)
Comment (shepeliev) [reply firebase-common/build.gradle.kts]: Fixed

File: firebase-firestore/build.gradle.kts (hunk 2 above)
Comment (shepeliev) [reply firebase-firestore/build.gradle.kts]: Fixed

**Review by julianfalcionelli — APPROVED** [review]

**Comment by suntrix** [comment]
@nbransby possible to get this reviewed/merged?

**Review by nbransby — APPROVED** [review]

**Pull request merged**